### PR TITLE
improve batch query

### PR DESCRIPTION
**Pull request by pxp928**
# Description of the PR

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If GraphQL schema is changed, GraphQL client updates/additions have been made
- [ ] If OpenAPI spec is changed, `make generate` has been run
- [ ] If ent schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/assembler/backends/ent/backend/search.go [hunk 1]
```diff
@@ -310,12 +310,41 @@ func (b *EntBackend) BatchQueryPkgIDCertifyVuln(ctx context.Context, pkgIDs []st
...
+		PkgID          uuid.UUID `json:"package_id"`
+		VulnID         uuid.UUID `json:"vulnerability_id"`
+		LastScanTimeDB time.Time `json:"max"`
+	}
+
+	var aggPredicates []predicate.CertifyVuln
+	aggPredicates = append(aggPredicates, certifyvuln.PackageIDIn(queryList...), certifyvuln.VulnerabilityIDNEQ(noVulnID))
+
+	// aggregate to find the latest timescanned for certifyVulns for list of packages
+	err := b.client.Debug().CertifyVuln.Query().
```
Comment (mrizzi) [inline pkg/assembler/backends/ent/backend/search.go:323]: I suppose `Debug()` supported the development but it's worth removing.
```suggestion
	err := b.client.CertifyVuln.Query().
```

File: pkg/assembler/backends/ent/backend/search.go [hunk 2]
```diff
@@ -344,15 +373,46 @@ func (b *EntBackend) BatchQueryPkgIDCertifyLegal(ctx context.Context, pkgIDs []s
...
+		PkgID             uuid.UUID `json:"package_id"`
+		DeclaredLicense   string    `json:"declared_licenses_hash"`
+		DiscoveredLicense string    `json:"discovered_licenses_hash"`
+		LastScanTimeDB    time.Time `json:"max"`
+	}
+
+	var aggPredicates []predicate.CertifyLegal
+	// aggregate to find the latest timescanned for certifyLegals for list of packages
+	aggPredicates = append(aggPredicates, certifylegal.PackageIDIn(queryList...), certifylegal.SourceIDIsNil())
+	err := b.client.Debug().CertifyLegal.Query().
```
Comment (mrizzi) [inline pkg/assembler/backends/ent/backend/search.go:386]: Same as above
```suggestion
	err := b.client.CertifyLegal.Query().
```

**Review by mrizzi — COMMENTED** [review]
Just some minors suggestions for cleaning the code.

File: pkg/assembler/backends/ent/backend/search.go (hunk 1 above)
Comment (pxp928) [reply pkg/assembler/backends/ent/backend/search.go]: ah yes i meant to remove that!

**Review by mrizzi — APPROVED** [review]
@pxp928 LGTM, thanks :+1:

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**